provider/resource_edge_deployment_service.go

```diff
@@ -42,7 +60,10 @@ func readEdgeDeploymentService(d *schema.ResourceData, m interface{}) error {
 func updateEdgeDeploymentBackends(d *schema.ResourceData, m interface{}) error {
 	pm := m.(providerMetadata)
 
-	return pm.Client.CreateOrUpdateEdgeDeploymentService(pm.Corp, d.Get("site_short_name").(string), d.Get("fastly_sid").(string))
+	return pm.Client.CreateOrUpdateEdgeDeploymentService(pm.Corp, d.Get("site_short_name").(string), d.Get("fastly_sid").(string), sigsci.CreateOrUpdateEdgeDeploymentServiceBody{
```

Comment:
@mattg-sigsci this might just be a typo in the function name (`updateEdgeDeploymentBackends`) but just to clarify, when should the user call `PUT /corps/{corpName}/sites/{siteName}/edgeDeployment/{fastlySID}` vs `PUT /corps/{corpName}/sites/{siteName}/edgeDeployment/{fastlySID}/backends`?

If the user changes the Fastly SID in their TF and runs `apply`, I think we need to call `CreateOrUpdateEdgeDeploymentService`, right?

Reply:
/backends is only called when you need to synchronize backend changes from your VCL service

Reply:
I guess I'm wondering which endpoint should be called when changes are made to a `sigsci_edge_deployment_service.site_short_name` or `sigsci_edge_deployment_service.fastly_sid`? I believe when those are changed, the TF provider will call the `Update` method.

Reply:
I think when either `short_name` or `fastly_sid` are changed CreateOrUpdateEdgeDeploymentService should be run as new NGWAF backends will need to be created and VCL inserted.